### DISCUSSION TITLE
Number of lines on cartesian grid to label domain and range are sometimes quite odd

### DISCUSSION THREAD

**Discussion by ToeJammm** [recharts/recharts#4696] [Q&A]
I've been working with a graph that has built in domain and range customization. Some ranges distribute the lines evenly, and some don't (even if it would be possible to). For example, my range in the picture below is 0-0.75. Each increment could be 0.25 greater than the last, but for some reason it goes to 0.2...0.4...and then 0.75 which is very unappealing to look at on a graph. Is there any  way to automate the equal distribution of tick marks in both x and y axis?
![Screenshot 2024-06-24 130631](https://github.com/recharts/recharts/assets/114703597/6fad3833-2601-46b3-8da3-7771ea9cd455)

**Comment by ckifer** [comment]
you can calculate ticks yourself if you want and supply them using the `ticks` array. 

`ticks={[0, 0.25, 0.5, 0.75]}`

You can try to use `interval="equidistantPreserveStart"` but that will probably remove the 0.75 at the top (preserving the start and not the end).

If you can make a sandbox (maybe the one we used before?) I can try to see whats going on

**Reply by ToeJammm** [reply]
I have recreated the issue I am having in the sandbox you forked for me

**Reply by ckifer** [reply]
You need to fork it, change it, make it public, and link it back here in order for me to see it. The one you linked me in the previous issue I couldn't access

**Reply by ToeJammm** [reply]
using the ticks variable made it look nice, I may just create a function to calculate even spacing

**Reply by ToeJammm** [reply]
https://codesandbox.io/p/sandbox/composed-chart-with-best-fit-forked-l84qrq?file=%2Fsrc%2FApp.tsx%3A25%2C35

**Reply by ckifer** [reply]
Yeah ideally recharts does this for you, but sometimes the calculations are different than people want. Will try to improve them after 3.0 is out

**Reply by ToeJammm** [reply]
Sounds like a plan. Recharts is still my favorite so far.

**Reply by ckifer** [reply]
https://github.com/recharts/recharts/pull/4453

This PR probably addresses the reason the calculation is off. But it comes with a cost/some weird API implications hence why it isn't merged yet.

Edit: on second thought maybe it isn't, but the hesitation to change things remains the same 😅
